lib/beautiful/xresources.lua

```diff
@@ -70,13 +70,21 @@ local function get_screen(s)
     return s and screen[s]
 end
 
---- Get global or per-screen DPI value falling back to xrdb.
--- @tparam[opt] integer|screen s The screen.
+--- Get global or per-screen DPI value, falling back to xrdb.
+-- @screen[opt] s The screen.
 -- @treturn number DPI value.
 function xresources.get_dpi(s)
-    s = get_screen(s)
-    if dpi_per_screen[s] then
-        return dpi_per_screen[s]
+    if s then
+        s = get_screen(s)
+        if dpi_per_screen[s] then
+            return dpi_per_screen[s]
+        end
+        local width = 0
+        for _,v in pairs(s.outputs) do
+            width = width + v.mm_width
+        end
+        local inch = width / 25.4
+        return math.floor(s.workarea.width / inch)
```

Comment:
This now ignores `Xft.dpi` even though the documentation says otherwise. Also, it's wrong. Adding the width of outputs is wrong.

Imagine clone mode: Multiple monitors (= outputs) display the same part of the screen. This is the case where a screen has multiple outputs. Of course adding their width together makes no sense here. The only possible thing might be to pick one of them arbitrarily, and that's bad ("arbitrary").

(Oh and the 25.4 magic number might need an explanation)